projects/openapi-utilities/src/openapi3/group-diff.ts

```diff
@@ -231,44 +293,99 @@ function normalizeRequiredDiff(
 
 export function groupDiffsByEndpoint(
```

Comment:
had to update this to group rules together with the same node - this impacts:
- the rule result renderer (still used by `lint`)
- the json changelog (i.e. needed to update the json changelog to check the diff count)
- the terminal changelog (which was completely rewritten)